packages/dds/map/src/mapKernel.ts

```diff
@@ -546,10 +604,18 @@ export class MapKernel {
             if (local) {
                 assert(localOpMetadata !== undefined,
                     0x014 /* `pendingMessageId is missing from the local client's ${op.type} operation` */);
-                const pendingMessageId = localOpMetadata as number;
-                const pendingKeyMessageId = this.pendingKeys.get(op.key);
-                if (pendingKeyMessageId === pendingMessageId) {
-                    this.pendingKeys.delete(op.key);
+                const pendingMessage = localOpMetadata as IMapKeyLocalOpMetadata;
+                const pendingMessageIds = this.pendingKeys.get(op.key);
+                if (pendingMessageIds !== undefined) {
+                    const index = pendingMessageIds.findIndex(
```

Comment:
do we actually need to walk the array or can asserts it's the first element? then we could just do pendingMessageIds.unshift() which also saves us the slice below

Reply:
I think it should always be ack'ed in order. I was worried that there might be some exceptions to this though, but it would have to know about how map works to be correct.

Reply:
On resubmit, won't there end up being two pendingMessageIds for the same message with the first one never being ack'ed?

Reply:
I removed the old pending message tracking when resubmitting.